### PR TITLE
Assorted code cleanups

### PR DESCRIPTION
**Pull request by kholia**
This is related to https://github.com/magnumripper/JohnTheRipper/issues/3001.

I would like to get some feedback before I start changing other files.

Changes,

1. Use the new OpenMP guidelines. Actual OpenMP tuning is left for a new PR.

2. I am using the `newline after variable declarations` rule from the Linux kernel world.

3. In our "format defines block", I am using spaces instead of tabs as spaces work well with every editor.

4. System headers on top, OpenMP a line below, and then rest of the headers.

5. Improved comment style, and formatting a bit.

Update: Not all formats have the same level of cleanups. Doing so would probably take days! This did not occur to me when I started this task.

### CODE REVIEW

**Comment by jfoug** [comment]
I know this sounds strange, but with all of these code 'cleanups' going on, and with only a subset of files being done, I wonder if it is time to add specific lines in the header comment, listing what code cleanup versions have been done?

We could version based upon the PR (or bug) number, Yes, I know @solardiz lists things will outliive GH, but that does not matter.  The number is simply a number.  We then create a cleanup document, and list just WHAT each cleanup number means.  That way, we know which source have been done, which ones not.  then if we later get newer developers, we always have something to let them play with.  Thoughts?  

So it could be something like this:

```c
/*
 *   copywrite no one, free for all to use
 *
 *   code cleaned #3013 jfoug, Dec 2017
 *   code cleaned #3176 magnum, Feb 2018
 */

$> cat ./doc/code_cleanup_levels.txt
These are specific instances where the JtR developers have deside to change / cleanup the existing code.

#3013
    Use the new OpenMP guidelines. Actual OpenMP tuning is left for a new PR.
    I am using the newline after variable declarations rule from the Linux kernel world.
    In our "format defines block", I am using spaces instead of tabs as spaces work well with every editor.
    System headers on top, OpenMP a line below, and then rest of the headers.
    Improved comment style, and formatting a bit.

#3176
    commenting style changed to (something)
    replace unsigned long long with uint64_t
    replace long long with int64_t
    replace unsigned int with unsigned
    ....
```

**Comment by jfoug** [comment]
Note, this would also work to satisfy some of the copyright issues.  We could write a single copyright disclaimer file about the cleanup type tasks.  Put everyone's name that does the cleanup into that file, disclaiming ownership of minor changes done in code cleanup.  Then the simple line in the source header would be a point back and a reference to that umbrella disclaimer.

@solardiz what do you think of this idea?  I know there have been YEARS of code cleanups, all undocumented, other than scattered github/john-dev posts.  But do you see value in doing this?

**Comment by magnumripper** [comment]
> I wonder if it is time to add specific lines in the header comment, listing what code cleanup versions have been done?

This works in theory but is bound to fail IRL, I'm pretty sure.

A cool property of Dhiru's formats is they are so much copy-pasted you can fix an issue in several hundred files with just a `git grep -l 'this'  | xargs sed -ri 's/this/that/g'`.

**Comment by magnumripper** [comment]
@kholia looks fine to me (glanced through a couple screens of similar fixes - and your description in OP looks fine to me). @solardiz are you happy with this OpenMP init (so we don't commit hundreds of fixes with a detail missing)?

File: src/episerver_fmt_plug.c [hunk 1]
```diff
@@ -134,6 +134,7 @@ static struct custom_salt {
 	int version;
 	unsigned char esalt[18 + 1]; /* base64 decoding, 24 / 4 * 3 = 18 */
 } *cur_salt;
+
 #if defined(_OPENMP) || defined(SIMD_COEF_32)
 static int omp_t = 1;
```
Comment (solardiz) [inline src/episerver_fmt_plug.c]: Does this format need `omp_t` to be global to it? Why?

File: src/has160_fmt_plug.c [hunk 2]
```diff
@@ -73,10 +73,13 @@ static void init(struct fmt_main *self)
 #ifdef _OPENMP
 	int omp_t;
 
-	omp_t = omp_get_max_threads();
-	self->params.min_keys_per_crypt *= omp_t;
-	omp_t *= OMP_SCALE;
-	self->params.max_keys_per_crypt *= omp_t;
+	int omp_t = omp_get_max_threads();
```
Comment (solardiz) [inline src/has160_fmt_plug.c]: Looks like duplicate definition of `omp_t` - does this even build?

File: src/openbsdsoftraid_fmt_plug.c [hunk 3]
```diff
@@ -97,10 +96,13 @@ static struct custom_salt {
...
-	self->params.min_keys_per_crypt *= omp_t;
-	omp_t *= OMP_SCALE;
-	self->params.max_keys_per_crypt *= omp_t;
+	int omp_t = omp_get_max_threads();
+
+	if (omp_t > 1) {
+		self->params.min_keys_per_crypt *= omp_t;
+		omp_t *= OMP_SCALE;
+		self->params.max_keys_per_crypt *= omp_t;
+	}
```
Comment (solardiz) [inline src/openbsdsoftraid_fmt_plug.c]: Somehow no syntax highlighting here - maybe unbalanced braces somewhere before?

File: src/radmin_fmt_plug.c [hunk 4]
```diff
@@ -172,16 +182,17 @@ static int cmp_exact(char *source, int index)
...
+	// and is null padded up to 100 bytes. We simply clean up prior buffer,
+	// up to element 99, but that element will never be written to.
 	if (cp < &saved_key[index][99])
 		while (*cp) *cp++ = 0;
 }
 
 static char *get_key(int index)
 {
-	// assured null teminated string.  Just return it.
+	// Assured null teminated string, just return it.
```
Comment (solardiz) [inline src/radmin_fmt_plug.c]: NUL

File: src/episerver_fmt_plug.c (hunk 1 above)
Comment (kholia) [reply src/episerver_fmt_plug.c]: Yes, it needs to be global. The `set_salt` is making use of `omp_t` variable.

File: src/has160_fmt_plug.c (hunk 2 above)
Comment (kholia) [reply src/has160_fmt_plug.c]: Fixed now. Yes, the build did succeed earlier.

File: src/openbsdsoftraid_fmt_plug.c (hunk 3 above)
Comment (kholia) [reply src/openbsdsoftraid_fmt_plug.c]: The code seems to be fine. I checked it again, and couldn't spot anything problematic.

File: src/radmin_fmt_plug.c (hunk 4 above)
Comment (kholia) [reply src/radmin_fmt_plug.c]: Ack. Should I use `NULL` or simply `NUL`? I have not seen usage of `NUL` previously, hence the question.

**Comment by kholia** [comment]
Here is the semantic patch that I used for the OpenMP stuff,

```
$ cat modernize-2.cocci 
@@
identifier omp_t;
identifier self;
@@
- omp_t = omp_get_max_threads();
- self->params.min_keys_per_crypt *= omp_t;
- omp_t *= OMP_SCALE;
- self->params.max_keys_per_crypt *= omp_t;
+ int omp_t = omp_get_max_threads();
+
+ if (omp_t > 1) {
+ 	self->params.min_keys_per_crypt *= omp_t;
+ 	omp_t *= OMP_SCALE;
+ 	self->params.max_keys_per_crypt *= omp_t;
+ }
```

**Comment by kholia** [comment]
I think I am done with the changes for now. This PR should be ready for another round of review. I am also double-checking the final changes now.

**Comment by kholia** [comment]
I am quite worried about merging such a big PR. I guess this will teach me to break such stuff into separate PRs next time.

Merging!

File: src/axcrypt_fmt_plug.c [hunk 5]
```diff
@@ -58,9 +61,9 @@ john_register_one(&fmt_axcrypt);
 
 static struct fmt_tests axcrypt_tests[] = {
 	/*
-		formats can be:
-		$axcrypt$*version*iterations*salt*wrappedkey
-		$axcrypt$*version*iterations*salt*wrappedkey*key-file
+	 * Formats can be,
```
Comment (frank-dittrich) [inline src/axcrypt_fmt_plug.c]: @kholia I think this change was wrong, you should have kept the `:` instead of changing it into a `,`.

Comment (kholia) [reply src/axcrypt_fmt_plug.c]: I believe that both `:` and `,` are OK. I don't remember why I made this change though.

**Pull request merged**